### PR TITLE
Adds `includes` function

### PR DESCRIPTION
**Pull request by SirJayEarn**

### CODE REVIEW

File: mapping.md [hunk 1]
```diff
@@ -56,4 +55,5 @@
 | `times`        | `times`        | `times`             |
 | `toPairs`      | `toPairs`      | `toPairs`           |
 | `type`         | `-`            | `type`              |
-| `uniq`         | `uniq`         | `uniq`              |
\ No newline at end of file
+| `uniq`         | `uniq`         | `uniq`              |
+| `includes`     | `includes`     | `includes`          |
```
Comment (sobstel) [inline mapping.md]: I think this is sorted alphabetically.

File: src/includes.test.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      expect(includes([1, 2, 3, 4, 5], 6)).toBe(false);
+    });
+
+    it('works with string arrays', () => {
+      expect(includes(['a', 'b', 'c', 'd', 'e'], 'c')).toBe(true);
+      expect(includes(['a', 'b', 'c', 'd', 'e'], 'f')).toBe(false);
+    });
+
+    it('works with objects', () => {
+      expect(includes({ a: 1, b: 2, c: 3 }, 'a')).toBe(true);
```
Comment (sobstel) [inline src/includes.test.ts]: This is not how lodash `includes` works. It checks for value, not for a key.

File: src/includes.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import { purry } from './purry';
+import { _reduceLazy } from './_reduceLazy';
+import { equals } from './equals';
+
+export type Includable<T> = T[] | Object | string;
```
Comment (sobstel) [inline src/includes.ts]: I'm not sure if it should support both object and string. As far as I saw all the other functions support either object OR string, so it's a bit inconsistent wit the rest of the lib.

**Comment by lstkz** [comment]
Can we support only strings (without regex), and arrays?

File: src/includes.test.ts (hunk 2 above)
Comment (SirJayEarn) [reply src/includes.test.ts]: Ok I see, so checking keys would rather be the job of `has`?

File: src/includes.ts (hunk 3 above)
Comment (SirJayEarn) [reply src/includes.ts]: Idk either - so what should `includes` handle? Only objects and arrays? Will there be a separate function for checking strings?

**Comment by SirJayEarn** [comment]
I'm fine with only supporting strings and arrays. But than it's not compatible with lodashs includes

File: src/includes.test.ts (hunk 2 above)
Comment (SirJayEarn) [reply src/includes.test.ts]: Now checking object values

**Comment by lstkz** [comment]
We don't have to replicate lodash in 100%.
`R.includes` should simply wrap the original `Array.includes`, and `String.includes` function.

> Why not using regex?

It can be confusing, the provided string must be a valid regex.
For example:
`R.includes('users (0)', '0)')`
throws an error

**Comment by SirJayEarn** [comment]
> We don't have to replicate lodash in 100%.
> `R.includes` should simply wrap the original `Array.includes`, and `String.includes` function.
> 
> > Why not using regex?
> 
> It can be confusing, the provided string must be a valid regex.
> For example:
> `R.includes('users (0)', '0)')`
> throws an error

Ah ok, I see. I'll change that

File: src/includes.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    return _arrayIncludes(item, source);
+  } else if(typeof item === 'string') {
+    return _stringIncludes(item, source);
+  } else {
+    return false;
+  }
+}
+
+function _arrayIncludes<T>(item: T, arr: T[]) {
+  return arr?.some(a => equals(a, item)) ?? false;
```
Comment (lstkz) [inline src/includes.ts]: - `arr?.` is not needed, it should be `arr.`
- `?? false` is not needed
- no partial matching, it should call `arr.includes`

File: src/includes.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+}
+
+function _stringIncludes(item: string, source: string) {
+  return source?.includes(item) ?? false;
```
Comment (lstkz) [inline src/includes.ts]: - similar to above, remove `??`, and `source?.` -> `source.`

File: src/includes.ts [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  return purry(_includes, arguments);
+}
+
+function _includes<T>(source: Includable<T>, item: T): boolean {
+  if (Array.isArray(source)) {
+    return _arrayIncludes(item, source);
+  } else if(typeof item === 'string') {
+    return _stringIncludes(item, source);
+  } else {
+    return false;
```
Comment (lstkz) [inline src/includes.ts]: if the type is not supported, typescript will report an error
instead of `return false`, we can throw an error

**Comment by lstkz** [comment]
Closing because of inactivity.

**Pull request closed without merging**